File: Domain/Post.cs
using System;

namespace Domain
{
    public class Post
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public DateTime Date { get; set; }
        public string Content { get; set; }
        public string AuthorId { get; set; }
        public AppUser Author { get; set; }
    }
}

File: client-app/src/app/models/post.ts
export interface Post {
    id: string;
    title: string;
    date: Date | null;
    content: string;
}

export class Post implements Post {
    constructor(init?: PostFormValues) {
        Object.assign(this, init);
    }
}

export class PostFormValues {
    id?: string = undefined;
    title: string = '';
    content: string = '';
    date: Date | null = null;

    constructor(post?: PostFormValues) {
        if (post) {
            this.id = post.id;
            this.title = post.title;
            this.content = post.content;
            this.date = post.date;
        }
    }
} 